### PR TITLE
[Opaque Pointer] Remove macros in CMakeLists and final clean up

### PR DESCRIPTION
**Pull request by jsji**
- Remove *ENABLE_OPAQUE_POINTERS macros
- Clean up of options
- Port tests

### CODE REVIEW

**Review by smanna12 — APPROVED** [review]
CFE changes LGTM

File: llvm/CMakeLists.txt [hunk 1]
```diff
@@ -905,18 +905,6 @@ if(NOT LLVM_ENABLE_NEW_PASS_MANAGER)
                       " no longer supported.")
 endif()
 
-set(DPCPP_ENABLE_OPAQUE_POINTERS TRUE CACHE BOOL
-   "Enable opaque pointers by default.")
-if (DPCPP_ENABLE_OPAQUE_POINTERS)
-  add_definitions("-DENABLE_OPAQUE_POINTERS=1")
-  add_definitions("-DINTEL_SYCL_OPAQUEPOINTER_READY=1")
```
Comment (sarnex) [inline llvm/CMakeLists.txt:912]: Do we need to remove this line too https://github.com/intel/llvm/blob/7a9c466c19230b3b82c7ce58366cc4815aeb6c6d/llvm/cmake/modules/TableGen.cmake#L7

**Review by sarnex — APPROVED** [review]
tools and esimd LGTM besides one question

File: sycl/test-e2e/ESIMD/vadd_usm_opqptr.cpp [hunk 2]
```diff
@@ -5,7 +5,7 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
-// RUN: %{build} -Xclang -opaque-pointers -o %t.out
+// RUN: %{build} -o %t.out
```
Comment (sarnex) [inline sycl/test-e2e/ESIMD/vadd_usm_opqptr.cpp:8]: sorry another nit while we're here, we should actually just delete this. all this does is include another test `vadd_usm.cpp` and that will be running with opaque pointers now, so we're running the same test twice

File: llvm/CMakeLists.txt (hunk 1 above)
Comment (jsji) [reply llvm/CMakeLists.txt:912]: Good catch. Removed.

File: sycl/test-e2e/ESIMD/vadd_usm_opqptr.cpp (hunk 2 above)
Comment (jsji) [reply sycl/test-e2e/ESIMD/vadd_usm_opqptr.cpp:8]: Sounds good.

**Review by sarnex — APPROVED** [review]
lgtm with fixes, thanks!

**Review by mdtoguchi — APPROVED** [review]
OK for driver tests - but @asudarsa should take a look as they are tools tests.

File: libclc/utils/libclc-remangler/LibclcRemangler.cpp [hunk 3]
```diff
@@ -778,11 +778,6 @@ class LibCLCRemangler : public ASTConsumer {
   void Initialize(ASTContext &C) override {
     ASTCtx = &C;
     SMDiagnostic Err;
-#if SPIRV_ENABLE_OPAQUE_POINTERS || !defined(__SPIR__)
-    LLVMCtx.setOpaquePointers(true);
-#else
-    LLVMCtx.setOpaquePointers(false);
```
Comment (jchlanda) [inline libclc/utils/libclc-remangler/LibclcRemangler.cpp:784]: The default is that context has it set to `false`, so there is no need for `setOpaquePointers`, right?

**Review by jchlanda — APPROVED** [review]
Remangler 👍

**Review by againull — APPROVED** [review]

**Comment by jsji** [comment]
Ping @asudarsa @intel/dpcpp-spirv-reviewers .

**Review by asudarsa — APPROVED** [review]
SPIR-V and Tools changes look good.

Thanks

**Comment by jsji** [comment]
Thanks all for the review! @intel/llvm-gatekeepers Can we get this merged? Thanks.

**Pull request merged**